### PR TITLE
code optimizations

### PR DESCRIPTION
**Pull request by iossocket**
# code optimizations

Some optimizations to improve user experience and performance, fixed all the chrome console warnings. 
Bench benchmarks reference: https://vercel.com/docs/speed-insights/metrics

Before:
image
After:
image

## Types of change

- [ ] Feature
- [ ] Bug
- [x] Enhancement

### CODE REVIEW

File: packages/nextjs/app/configure/_components/DownloadContracts.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 "use client";
 
 import { useProvider } from "@starknet-react/core";
-import React, { useState } from "react";
+import React, { useCallback, useState } from "react";
```
Comment (metalboyrick) [inline packages/nextjs/app/configure/_components/DownloadContracts.tsx:4]: just wondering if we can remove "import React" ? wdyt?

Comment (iossocket) [reply packages/nextjs/app/configure/_components/DownloadContracts.tsx]: good point, yes, we can

File: packages/nextjs/hooks/scaffold-stark/useScaffoldContract.ts [hunk 2]
```diff
@@ -12,16 +10,7 @@ export const useScaffoldContract = <TContractName extends ContractName>({
...
-
-  const { targetNetwork } = useTargetNetwork();
-  const { account } = useAccount();
-  const publicNodeUrl = targetNetwork.rpcUrls.public.http[0];
-
-  const publicClient = useMemo(() => {
-    return new RpcProvider({
-      nodeUrl: publicNodeUrl,
-    });
-  }, [publicNodeUrl]);
```
Comment (metalboyrick) [inline packages/nextjs/hooks/scaffold-stark/useScaffoldContract.ts:24]: are these unused code?

**Review by metalboyrick — COMMENTED** [review]
some more minor comments and i think all good from my side

File: packages/nextjs/hooks/scaffold-stark/useScaffoldContract.ts (hunk 2 above)
Comment (iossocket) [reply packages/nextjs/hooks/scaffold-stark/useScaffoldContract.ts:24]: replaced by `const { provider: publicClient } = useProvider();`

**Review by Nadai2010 — APPROVED** [review]

**Comment by Nadai2010** [comment]
Hi Team, very good work @iossocket , all test/review and is all ok with me.

**Review by 0xquantum3labs — APPROVED** [review]

**Pull request merged**